module/VuFindSearch/src/VuFindSearch/Backend/Blender/Response/Json/RecordCollection.php

```diff
@@ -0,0 +1,441 @@
+     * Mappings configuration
+     *
+     * @var array
+     */
+    protected $mappings;
+
+    /**
+     * Backends to be used for initial results
+     *
+     * @var array
+     */
+    protected $initialResults;
+
+    /**
+     * Any errors encountered
+     *
+     * @var array
+     */
+    protected $errors = [];
+
+    /**
+     * Constructor
+     *
+     * @param \Laminas\Config\Config $config   Configuration
+     * @param array                  $mappings Mappings configuration
+     */
+    public function __construct($config = null, $mappings = [])
+    {
+        $this->config = $config;
+        $this->mappings = $mappings;
+        $this->response = static::$template;
+        $this->initialResults = isset($this->config->Blending->initialResults)
+            ? $this->config->Blending->initialResults->toArray()
+            : [];
+    }
+
+    /**
+     * Initialize blended results
+     *
+     * Creates a record list from 0 to $limit
+     *
+     * @param array $collections Array of record collections
+     * @param int   $limit       Result limit
+     * @param int   $blockSize   Blending block size
+     * @param int   $totalCount  Total result count
+     *
+     * @return array Remaining records keyed by backend identifier
+     */
+    public function initBlended(
+        array $collections,
+        int $limit,
+        int $blockSize,
+        int $totalCount
+    ): array {
+        $this->response = static::$template;
+        $this->response['response']['numFound'] = $totalCount;
+        $this->rewind();
+
+        if (!$collections) {
+            return [];
+        }
+
+        $backendRecords = [];
+        foreach ($collections as $backendId => $collection) {
+            $records = $collection->getRecords();
+            $label = $this->config->Backends[$backendId];
+            foreach ($records as $record) {
+                $record->setSourceIdentifiers(
+                    $record->getSourceIdentifier(),
+                    $backendId
+                );
+                if ($label) {
+                    $record->addLabel($label, 'source');
+                }
+                $backendRecords[$backendId][] = $record;
+            }
+
+            foreach ($collection->getErrors() as $error) {
+                if (is_string($error) && $label) {
+                    $error = [
+                        'msg' => '%%error%% -- %%label%%',
+                        'tokens' => [
+                            '%%error%%' => $error,
+                            '%%label%%' => $label
+                        ],
+                        'translate' => true,
+                        'translateTokens' => true,
+                    ];
+                }
+                $this->addError($error);
+            }
+        }
+
+        $records = [];
+        $backendIds = array_keys($backendRecords);
+        // Filter out unavailable backends from initial results source list:
+        $initialResults = array_values(
```

Comment:
I'm probably not reading carefully enough, but is there a reason we can't just use `array_diff` here?

Reply:
Yes, `array_intersect`, done.